core/src/main/java/org/mapstruct/SubClassMapping.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright MapStruct Authors.
+ *
+ * Licensed under the Apache License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package org.mapstruct;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Repeatable;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * Configures the mapping to handle hierarchy of the source type.
+ * <p>
+ * The subclass to be mapped is to be specified via {@link #sourceClass()}.
+ * The subclass to map to is to be specified via {@link #targetClass()}.
+ * It currently is also required that a mapping method exist that has the {@link #targetClass()}
+ * as the return type and the {@link #sourceClass()} as the only argument.
```

Comment:
Let's try to get rid of this requirement. I've left bunch of comments around that I think will help for this.

Reply:
The existing mapping is no longer required, but I'm still not certain about the mappings that use the update route. That might need some more work, but let's see how far we'll get.

Reply:
I was more referring to the fact about "required that a mapping methods exists". That is not needed, since we can create an automatic method for it. We can mention that it does not support update methods though, no need to look into that for this PR

Reply:
I was referring to the same thing with the remark 'The existing mapping is no longer required'. :)
About the no support for update methods bit, I will add it (if not already there) to the documentation contained in the PR.

Note to self, this post requires a thumbs up when I've updated the documentation.